resources/classdetail/disk/compute_disk.go

```diff
@@ -0,0 +1,117 @@
+package disk
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+)
+
+type diskJSON struct {
+	Name              string
+	Region            string
+	Zone              string
+	DefaultDiskSizeGb string
+	ValidDiskSize     string
+}
+
+type disk struct {
+	Type           string
+	Region         string
+	Zone           string
+	DefaultSizeGiB int64
+	MinSize        int64
+	MaxSize        int64
+}
+
+// Disks are stored first by type, then by zone/region.
+var diskTypes map[string]map[string]*disk
+
+func convertToDisk(d diskJSON) (*disk, error) {
+	final := &disk{Type: d.Name, Region: d.Region, Zone: d.Zone}
+
+	def, err := strconv.ParseInt(d.DefaultDiskSizeGb, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	final.DefaultSizeGiB = def
+
+	// Separate min and max strings and remove trailing "GB" before conversion.
+	i := strings.LastIndex(d.ValidDiskSize, "-")
+	if i < 0 {
+		return nil, fmt.Errorf("invalid disk size interval format")
+	}
+
+	min, err := strconv.ParseInt(d.ValidDiskSize[:i-2], 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	final.MinSize = min
+
+	max, err := strconv.ParseInt(d.ValidDiskSize[i+1:len(d.ValidDiskSize)-2], 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	final.MaxSize = max
+
+	return final, nil
+}
+
+func readDiskInfo() error {
```

Comment:
Nit: This function is reading info about disk types, so maybe `readDiskTypes` is a better name.